### ISSUE TITLE
Experimenter testing command doesn't fetch experiments from overridden server endpoint (like staging server)

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/android-components#4779]
In setting up the first Mobile Experimenter experiment with [ETP on Fenix](https://github.com/mozilla-mobile/fenix/issues/5651), we can't seem to use the records from the staging server.

### Acceptance Criteria
- [ ] Overriding endpoint in adb testing instructions fetches experiments from the correct endpoint
- [ ] Includes a test
- [ ] Timeline: in the next few weeks (**done before 11/8**), before any upcoming Fenix experiments
- [ ] The experiment testing (e.g. changing an endpoint) doesn't make assumptions about where experiments are init-ed (specifically, whether experiments can be init-ed in MainApplication).
- [ ] ^If not possible, or too difficult, update documentation to be clear about any constraints testing creates.  

Even though we're overriding the endpoint to use staging, it's still fetching from prod.
I don't see [a test testing pulling experiments from the updated endpoint](https://github.com/mozilla-mobile/android-components/blob/master/components/service/experiments/src/test/java/mozilla/components/service/experiments/debug/ExperimentsDebugActivityTest.kt#L101).

staging: https://settings-writer.stage.mozaws.net/v1
prod: https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/mobile-experiments/records?_since=1566233826256
command: `adb shell am start -n org.mozilla.fenix/mozilla.components.service.experiments.debug.ExperimentsDebugActivity  --ez updateExperiments true  --es setKintoInstance "staging" --es overrideExperiment "fenix-etp-5651" --es branch “treatment_standard"`

```
2019-10-17 15:15:34.740 6476-6542/org.mozilla.fenix.debug I/experiments: Fetching experiments from https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/mobile-experiments/records?_since=1566233826256
2019-10-17 15:15:34.741 6476-6476/org.mozilla.fenix.debug I/ExperimentsDebugActivity: Changing Kinto instance to https://settings-writer.stage.mozaws.net/v1
2019-10-17 15:15:34.794 6476-6542/org.mozilla.fenix.debug D/experiments: Fetched experiments from https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/mobile-experiments/records?_since=1566233826256:
    {"data":[]}
2019-10-17 15:15:34.795 6476-6542/org.mozilla.fenix.debug I/experiments: Experiments update from server: ExperimentsSnapshot(experiments=[Experiment(id=fenix-test-2019-08-05, description=End-to-end test of the Mobiles Experiments platform for Firefox Preview., lastModified=1566233826256, schemaModified=1566228455761, buckets=Buckets(start=0, count=667), branches=[Branch(name=a, ratio=1), Branch(name=b, ratio=1), Branch(name=c, ratio=1)], match=Matcher(appId=^org.mozilla.fenix$, appDisplayVersion=null, appMinVersion=null, appMaxVersion=null, localeLanguage=null, localeCountry=null, deviceManufacturer=null, deviceModel=null, regions=[], debugTags=[]))], lastModified=1566233826256)
2019-10-17 15:15:34.799 6476-6542/org.mozilla.fenix.debug I/experiments: onExperimentsUpdated - no experiment currently active, looking for match
2019-10-17 15:15:59.725 6476-6476/org.mozilla.fenix.debug I/ExperimentsDebugActivity: Override to the following experiment/branch:fenix-etp-5651/treatment_standard
```

A few things to investigate:
- Is this dependent on where the experiments are initialized in the `MainApplication`? The `ExperimentsDebugActivity` needs the Application to have been initialized before it can run, so does this preclude having the experiment initialization in `MainApplication`? (The testing framework shouldn't require code structure changes.)
- Should the experiments be updated [after checking the endpoint](https://github.com/mozilla-mobile/android-components/blob/master/components/service/experiments/src/main/java/mozilla/components/service/experiments/debug/ExperimentsDebugActivity.kt#L79)?
- Is the updated endpoint in the configuration somehow not the same one [used to fetch the new experiments](https://github.com/mozilla-mobile/android-components/blob/master/components/service/experiments/src/main/java/mozilla/components/service/experiments/debug/ExperimentsDebugActivity.kt#L102)?

**Comment by ekager** [comment]
Seems like #4783 is the first problem

**Comment by liuche** [comment]
Since #4783 landed, @mythmon you should try verifying the first item! If it works that solves most of the problems.

**Comment by liuche** [comment]
We think there may be a Secure Proxy test that will get run in early December, so these fixes for override, testing should be ready by then, so we don't have to have the same high-pressure situation for QA again.

**Comment by ekager** [comment]
Hey @mythmon since we may be doing another experiment soon wondering if you have a chance to look at this? We want to make sure the next experiment goes a bit more smoothly :)

**Comment by travis79** [comment]
I believe I've finally had time to discover why this is happening.  The `ExperimentsDebugActivity` tool was resetting the source of the experiments, but it wasn't actually using that source when the update occurred.  

I was able to diagnose this more easily with the changes I made for PR #5822, which adds a callback mechanism to notify the app when the experiments were updated.  

I'm happy to commit to that PR the one line fix that takes care of this, or to put up a separate PR if that is preferred.
